extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authorization/BasicRoleBasedAuthorizer.java

```diff
@@ -90,7 +94,11 @@ public Access authorize(AuthenticationResult authenticationResult, Resource reso
     Map<String, BasicAuthorizerRole> roleMap = roleProvider.getRoleMap(name);
 
     if (roleNames.isEmpty()) {
-      return new Access(false);
+      LOG.info("No roles found for authorizer [%s] and user [%s]", name, authenticationResult.getIdentity());
+      return new Access(false, StringUtils.format("You don't have '%s' permission for Resource [%s:%s]",
+          action.name(),
+          resource.getType(),
+          resource.getName()));
```

Comment:
Is it safe to expose the name of the resource to the end user?
If the error message when a resource doesn't exist versus when a resource exists but the user is not authorized to access it is not the same, might lead to the user knowing which resources are present in the cluster.

Reply:
The PR doesn't change the error code in any way. We are appending the information to the error. I think it is safe to expose the name of the resource that they are trying to access.